lib/parallel_tests/pids.rb

```diff
@@ -0,0 +1,68 @@
+require 'tempfile'
+require 'singleton'
+require 'json'
+
+module ParallelTests
+  class Pids
+    attr_reader :file, :pids, :mutex
+
+    include Singleton
+
+    def initialize
+      @file = Tempfile.new('parallel_pids')
```

Comment:
might be simpler to pass in the `path` instead of generating and then extracting

Reply:
would be nice to have the api be `add/delete/count` ... maybe `.send` the rest and keep them private ... either way is fine I guess